File: apps/api/app/schemas/tickets.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, Field


class TicketCreate(BaseModel):
    subject: str = Field(
        ...,
        min_length=3,
        max_length=120,
        description="Short title of the support ticket",
    )
    description: str = Field(
        ...,
        min_length=10,
        max_length=2000,
        description="Full description of the customer issue",
    )
    priority: Literal["low", "medium", "high", "urgent"] = "medium"
    customer_email: str | None = None


class TicketResponse(BaseModel):
    id: str
    subject: str
    description: str
    priority: Literal["low", "medium", "high", "urgent"]
    status: Literal["open", "in_progress", "resolved", "closed"]
    created_at: datetime


class ErrorResponse(BaseModel):
    error: str
    detail: str

File: apps/web/src/types/tickets.ts
export type TicketStatus = "Open" | "In Progress" | "Resolved";
export type TicketPriority = "High" | "Medium" | "Low";

export interface Ticket {
  id: string;
  subject: string;
  customer: string;
  customerEmail: string;
  description: string;
  status: TicketStatus;
  priority: TicketPriority;
  created: string;
}

export interface CreateTicketPayload {
  subject: string;
  customerEmail: string;
  priority: TicketPriority;
  description: string;
}
